### PR TITLE
Keyboard

### PR DESCRIPTION
**Pull request by dawnerd**

### CODE REVIEW

File: lib/complete.js [hunk 1]
```diff
@@ -2,10 +2,14 @@
...
 import debounce from 'tinybounce';
+import { on, addClass, removeClass, toArray } from 'domassist';
 
 export default class Complete extends Domodule {
-  get showClass() {
-    return this.options.showClass || 'show';
+  constructor(el) {
+    super(el);
+
+    this.currentIndex = -1;
```
Comment (jgallen23) [inline lib/complete.js]: you can use postInit instead of constructor

File: lib/complete.js [hunk 2]
```diff
@@ -2,10 +2,14 @@
...
+    on(this.els.input, 'keyup', event => this.keyup(event));
```
Comment (jgallen23) [inline lib/complete.js]: oo, multiple events on a single element... hmm, I guess this is the best way to handle.

`on(this.els.input, 'keyup', this.keyup.bind(this))` works too

File: lib/complete.js [hunk 3]
```diff
@@ -84,6 +97,66 @@ export default class Complete extends Domodule {
 
     return output;
   }
+
+  highlightItem() {
+    const items = toArray(this.els.resultsContainer.querySelectorAll('li'));
```
Comment (jgallen23) [inline lib/complete.js]: can also do `toArray(find('li', this.els.resultsContainer))`

File: lib/complete.js [hunk 4]
```diff
@@ -84,6 +97,66 @@ export default class Complete extends Domodule {
...
+
+    if (this.currentIndex < 0) {
+      this.currentIndex = 0;
+    }
+
+    if (this.currentIndex >= items.length) {
+      this.currentIndex = items.length - 1;
+    }
+
+    items.forEach(item => {
```
Comment (jgallen23) [inline lib/complete.js]: just added a task for removeClass to take in array of elements, that will clean this up

File: lib/complete.js (hunk 2 above)
Comment (dawnerd) [inline lib/complete.js]: Ah, wasn't sure bind was supported well enough. Looks like it is, so I'll switch it.

File: lib/complete.js [hunk 5]
```diff
@@ -21,6 +25,7 @@ export default class Complete extends Domodule {
     }
     this.term = term;
     this.endpoint = options.endpoint || this.options.endpoint;
+    this.listClass = options.listClass || this.options.listClass;
```
Comment (jgallen23) [inline lib/complete.js]: we should build in some sort of defaults huh?

Comment (jgallen23) [inline lib/complete.js]: defaults were just added by @Belelros, we should impletment when domodule is released

**Pull request merged**